pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -2081,6 +2081,21 @@ func TestReconcileWithWhenExpressionsWithParameters(t *testing.T) {
 	if d := cmp.Diff(actualSkippedTasks, expectedSkippedTasks); d != "" {
 		t.Errorf("expected to find Skipped Tasks %v. Diff %s", expectedSkippedTasks, diff.PrintWantGot(d))
 	}
+
+	skippedTasks := []string{"hello-world-2"}
+	for _, skippedTask := range skippedTasks {
+		labelSelector := fmt.Sprintf("tekton.dev/pipelineTask=%s,tekton.dev/pipelineRun=test-pipeline-run-different-service-accs", skippedTask)
+		actualSkippedTask, err := clients.Pipeline.TektonV1beta1().TaskRuns("foo").List(metav1.ListOptions{
+			LabelSelector: labelSelector,
+			Limit:         1,
+		})
+		if err != nil {
+			t.Fatalf("Failure to list TaskRun's %s", err)
+		}
+		if len(actualSkippedTask.Items) != 0 {
+			t.Fatalf("Expected 0 TaskRuns got %d", len(actualSkippedTask.Items))
+		}
+	}
 }
 
 func TestReconcileWithWhenExpressionsWithTaskResults(t *testing.T) {
```

Comment:
@jerop I am expecting a pipeline task here which has reference to task result but the result does not match with what is expected i.e.

```
		tb.PipelineTask("task-result-mismatch", "task",
			tb.PipelineTaskWhenExpression("$(tasks.a-task.results.aResult)", selection.In, []string{"random"}),
		),
```

Reply:
And also please add a test, checking the pipeline skips the entire branch in case of a task result mismatch.

Reply:
@pritidesai the example with result reference doesn't match what is expected is already here (c-task): https://github.com/tektoncd/pipeline/pull/3197/files/906414229d999cd9c7e6a5401f937c0bb3d97e7f#diff-f7c8532cfe79319c0fff711a4979a6d6R2107

and it has a dependent (d-task) that's also skipped https://github.com/tektoncd/pipeline/pull/3197/files/906414229d999cd9c7e6a5401f937c0bb3d97e7f#diff-f7c8532cfe79319c0fff711a4979a6d6R2112